### DISCUSSION TITLE
Add Ingredient Sections to recipes

### DISCUSSION THREAD

**Discussion by Imrhien** [mealie-recipes/mealie#2527] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Some recipes are broken into sections, for example a salad might have the salad ingredients and the dressing ingredients. It can be confusing when such recipes are imported without the sections as you can mix up which ingredients are for which parts, or the recipe might say "combine all the dressing ingredients" and it's not clear which ingredients those are.

### Please provide a concise description of the feature that would resolve your issue.

In the same way that there are sections for recipe steps, the ability to insert sections when adding or parsing ingredients would solve the problem.

### Please consider and list out some caveats or tradeoffs made in your design decision

It may not be feasible to include sections when importing a recipe. 

For an initial release this may need to be a manual process.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Which version of Mealie are you using? This has been a feature for a while:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/e3d2ba36-f4d9-40c5-94ff-d2cee4e330e3)

Assuming you're on the 1.0.0 beta, while in edit mode, hit the context menu on a recipe step and click "Toggle Section":
![image](https://github.com/mealie-recipes/mealie/assets/71845777/da9ae6e9-13c6-4ed1-acf7-cde008c266aa)

**Reply by Imrhien** [reply]
I can see it now! Sorry for not noticing this feature!

Are there docs for how to use Mealie somewhere? And, if not - I'd be happy to work on them :)

**Reply by michael-genson** [reply]
No worries! There are docs, but additions are always welcome!

https://nightly.mealie.io/documentation/getting-started/introduction/

**Reply by michael-genson** [reply]
FYI ingredients have sections too, similar to steps

**Reply by JvdMaat** [reply]
They just don't import from a URL recipe creation.
Just like parsing the ingredients, you'll also need to toggle the section manually.
